File: tests/e2e/fixtures/code/sample.php
<?php
// PHP: classes, typed properties, arrays, string interpolation.
declare(strict_types=1);

namespace NativeFileEditor;

final class Note
{
    public function __construct(
        public readonly string $path,
        public readonly array $tags = [],
    ) {}

    public function size(): int
    {
        return filesize($this->path) ?: 0;
    }
}

/** @param Note[] $notes */
function groupByTag(array $notes): array
{
    $out = [];
    foreach ($notes as $note) {
        foreach ($note->tags as $tag) {
            $out[$tag][] = $note;
        }
    }
    echo "grouped " . count($out) . " tags\n";
    return $out;
}

File: tests/e2e/fixtures/code/sample.dart
// Dart: classes, null safety, async, string interpolation.
import 'dart:io';

class Note {
  final String path;
  final List<String> tags;
  const Note(this.path, [this.tags = const []]);

  Future<int> size() async => (await File(path).stat()).size;
}

Map<String, List<Note>> groupByTag(Iterable<Note> notes) {
  final out = <String, List<Note>>{};
  for (final n in notes) {
    for (final t in n.tags) {
      (out[t] ??= []).add(n);
    }
  }
  return out;
}

void main() async {
  final n = Note('a.md', ['x']);
  print('${n.path}: ${await n.size()} bytes');
}
